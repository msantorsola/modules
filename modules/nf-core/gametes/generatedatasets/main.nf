process GAMETES_GENERATEDATASETS {
    tag "$meta.id"
    label 'process_medium'


    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/gametes:2.1--py310h7cba7a3_0':
        'biocontainers/gametes:2.1--py310h7cba7a3_0' }"


    input:
    tuple val(meta), path(model)


    output:
    tuple val(meta), path($prefix) , emit: results
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def args2 = task.ext.args2 ?: ''
    def prefix = task.ext.prefix ?: "$meta.id"
    def VERSION = '2.1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.

    """
    gametes \\
        --modelInputFile $model \\
        $args \\
        --dataset \\
        "$args2 \\
        -o $prefix"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gametes: $VERSION
    END_VERSIONS
    """
}
