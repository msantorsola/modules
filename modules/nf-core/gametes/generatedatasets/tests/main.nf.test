nextflow_process {

    name "Test Process GAMETES_GENERATEDATASETS"
    script "../main.nf"
    process "GAMETES_GENERATEDATASETS"

    tag "modules"
    tag "modules_nfcore"
    tag "gametes"
    tag "gametes/generatedatasets"
    config "./nextflow.config"

    test("dataset - model") {


        when {
            process {
                """
                input[0] = [[ id:'test'],
                file("https://github.com/msantorsola/random_tests/main/model_h0.2_p0.3_a0.3_f0.4_quantile_1_Models.txt")]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.edmresults).get(0).get(1) ==~ ".*/*._EDM_*" }
                { assert snapshot(process.out.edmresults).get(0).get(1) ==~ ".*/*._EDM_*" }
                )
        }
    }
}

