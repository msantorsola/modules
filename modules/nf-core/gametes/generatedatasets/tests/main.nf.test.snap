{
    "dataset - model": {
        "content": [
            "test_EDM-1_1.txt:md5,d653439e5008ac8352047548ceb7138f"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-25T09:42:11.75718707"
    }
}